app/scripts/views/sub_panels.js

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'jquery',
+  'lib/promise',
+  'views/base',
+  'stache!templates/sub_panels'
+],
+function ($, p, BaseView, Template) {
+  'use strict';
+
+  function renderView(view) {
+    return view.render()
+      .then(function (shown) {
+        if (! shown) {
+          view.destroy(true);
+          return;
+        }
+        view.afterVisible();
+
+        return view;
+      });
+  }
+
+  var View = BaseView.extend({
+    template: Template,
+    className: 'sub-panels',
+
+    initialize: function (options) {
+      options = options || {};
+
+      this._initialSubView = options.initialSubView;
+      this._panelViews = options.panelViews || [];
+    },
+
+    showSubView: function (SubView) {
+      var self = this;
+      if (self._panelViews.indexOf(SubView) === -1) {
+        console.warn('Tried to show a view that is not a subpanel');
```

Comment:
This is the kind of error that, if we had a static type system, would cause the build to fail, i.e. we should never ship with this kind of misconfiguration. Perhaps it can be refactor so that there's never a discrepancy.